### ISSUE TITLE
Programs don't build with libtestdriver and `USE_PSA`

### ISSUE THREAD

**Issue by mpg** [Mbed-TLS/mbedtls#6101]
In the new `all.sh` component `component_test_psa_crypto_config_accel_hash_use_psa()` introduced in #6065 we only build the tests, not programs, because they would fail to build.

```
  CC    x509/req_app.c
In file included from ../../include/psa/crypto_struct.h:80,
                 from ../../include/psa/crypto.h:4058,
                 from ../../include/mbedtls/pk.h:44,
                 from ../../include/mbedtls/x509.h:29,
                 from ../../include/mbedtls/x509_csr.h:28,
                 from fuzz_x509csr.c:4:
../../include/psa/crypto_driver_contexts_primitives.h:45:10: fatal error: libtestdriver1/include/psa/crypto.h: No such file or directory
   45 | #include <libtestdriver1/include/psa/crypto.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```
Note: the programs build just fine in the similar component `component_test_psa_crypto_config_accel_hash()`, and the difference causing that seems to be enabling `MBEDTLS_USE_PSA_CRYPTO`.

This task is to investigate the reason for this failure and fix it, so that the component can uses `make all` not just `make test`.

Depends on #6065 being merged for merging the fix, but the investigation can begin before that by using the branch from #6065 locally - if a fix is found before #6065 is merged, it will be picked up there, otherwise the fix will be a follow-up PR.

**Comment by superna9999** [comment]
Adding `-I../../tests` fixes this issue, but another build error appears:
```
fuzz_privkey.c:(.text+0x496): undefined reference to `mbedtls_entropy_init'
```

**Comment by mpg** [comment]
> Adding `-I../../tests` fixes this issue

Ah, I was sure I had tried that, but apparently not, or not correctly, or I failed to notice it made the error change...

> but another build error appears:
> 
> ```
> fuzz_privkey.c:(.text+0x496): undefined reference to `mbedtls_entropy_init'
> ```

Looks like this fuzzing program has undocumented dependencies, probably the existing `defined(MBEDTLS_CTR_DRBG_C)` guards need to be completed with `&& defined(MBEDTLS_ENTROPY_C)` - and similarly for other programs with undocumented dependencies.
